From ccc39e78cbd018cffc6ec93de75542ada863abe0 Mon Sep 17 00:00:00 2001
From: Richard Guy Briggs <rgb@redhat.com>
Date: Mon, 22 Sep 2014 13:59:37 -0400
Subject: [kernel] audit: correct AUDIT_GET_FEATURE return message type

Message-id: <b23fd379a2a7393f06309d748d61a096fcd5c297.1409008421.git.rgb@redhat.com>
Patchwork-id: 96429
O-Subject: [RHEL7 PATCH] audit: correct AUDIT_GET_FEATURE return message type
Bugzilla: 1141791
RH-Acked-by: Eric Paris <eparis@redhat.com>
RH-Acked-by: Steve Grubb <sgrubb@redhat.com>
RH-Acked-by: Paul Moore <pmoore@redhat.com>

When an AUDIT_GET_FEATURE message is sent from userspace to the kernel, it
should reply with a message tagged as an AUDIT_GET_FEATURE type with a struct
audit_feature.  The current reply is a message tagged as an AUDIT_GET
type with a struct audit_feature.

This appears to have been a cut-and-paste-eo in commit b0fed40.

Posted upstream: https://www.redhat.com/archives/linux-audit/2014-August/msg00072.html
BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1141791
BREW: https://brewweb.devel.redhat.com/taskinfo?taskID=7993924

Reported-by: Steve Grubb <sgrubb@redhat.com>
Signed-off-by: Richard Guy Briggs <rgb@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/kernel/audit.c b/kernel/audit.c
index f6badb2..6129138 100644
--- a/kernel/audit.c
+++ b/kernel/audit.c
@@ -663,7 +663,7 @@ static int audit_get_feature(struct sk_buff *skb)
 
  seq = nlmsg_hdr(skb)->nlmsg_seq;
 
- audit_send_reply(NETLINK_CB(skb).portid, seq, AUDIT_GET, 0, 0,
+ audit_send_reply(NETLINK_CB(skb).portid, seq, AUDIT_GET_FEATURE, 0, 0,
     &af, sizeof(af));
 
  return 0;
-- 
1.7.1